### PR TITLE
[RDY] Subclass Staff

### PR DESCRIPTION
**Pull request by TheCycoONE**
Refactor staff.lua into separate Doctor, Handyman, Nurse, Receptionist classes, for better code separation.

### CODE REVIEW

File: CorsixTH/Lua/entities/humanoids/staff/doctor.lua [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+  -- if you overwork your Dr's then there is a chance that they can go crazy
+  -- when this happens, find him and get him to rest straight away
+  if self.attributes["fatigue"] then
+    if self.attributes["fatigue"] < 0.7 then
+      if self:isResting() then
+        self:setCrazy(false)
+      end
+    else
+      -- doctor can go crazy if they're too tired
+      if self.humanoid_class == "Doctor" then
```
Comment (TheCycoONE) [inline CorsixTH/Lua/entities/humanoids/staff/doctor.lua]: Remove this condition, redundant

File: CorsixTH/Lua/entities/humanoids/staff/handyman.lua [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  self:Staff(...)
+end
+
+function Handyman:leaveAnnounce()
+  local announcement_priority = AnnouncementPriority.High
+  self.world.ui:playAnnouncement("sack006.wav", announcement_priority)
+end
+
+function Handyman:die()
+  if self.task then
```
Comment (mugmuggy) [inline CorsixTH/Lua/entities/humanoids/staff/handyman.lua]: This self.task appears in 3 different places, could be refactored into another method.

Comment (TheCycoONE) [reply CorsixTH/Lua/entities/humanoids/staff/handyman.lua]: Will do

**Comment by TheCycoONE** [comment]
@mugmuggy I think you can merge if you are happy with my changes.

**Pull request merged**